src/components/modules/toolbar/index.ts

```diff
@@ -172,54 +227,80 @@ export default class Toolbar extends Module<ToolbarNodes> {
       this.enableModuleBindings();
     } else {
       this.destroy();
-      this.Editor.Toolbox.destroy();
+      this.toolboxInstance.destroy();
       this.Editor.BlockSettings.destroy();
       this.disableModuleBindings();
     }
   }
 
   /**
-   * Move Toolbar to the Current Block
+   * Move Toolbar to the passed (or current) Block
    *
-   * @param {boolean} forceClose - force close Toolbar Settings and Toolbar
+   * @param block - block to move Toolbar near it
    */
-  public move(forceClose = true): void {
-    if (forceClose) {
-      /** Close Toolbox when we move toolbar */
-      this.Editor.Toolbox.close();
-      this.Editor.BlockSettings.close();
-    }
-
-    const currentBlock = this.Editor.BlockManager.currentBlock.holder;
+  public moveAndOpen(block: Block = this.Editor.BlockManager.currentBlock): void {
+    /**
+     * Close Toolbox when we move toolbar
+     */
+    this.toolboxInstance.close();
+    this.Editor.BlockSettings.close();
 
     /**
      * If no one Block selected as a Current
      */
-    if (!currentBlock) {
+    if (!block) {
       return;
     }
 
+    this.hoveredBlock = block;
+
+    const targetBlockHolder = block.holder;
     const { isMobile } = this.Editor.UI;
-    const blockHeight = currentBlock.offsetHeight;
-    let toolbarY = currentBlock.offsetTop;
+    const renderedContent = block.pluginsContent;
+    const renderedContentStyle = window.getComputedStyle(renderedContent);
+    const blockRenderedElementPaddingTop = parseInt(renderedContentStyle.paddingTop, 10);
+    const blockHeight = targetBlockHolder.offsetHeight;
+
+    let toolbarY;
 
     /**
-     * 1) On desktop — Toolbar at the top of Block, Plus/Toolbox moved the center of Block
-     * 2) On mobile — Toolbar at the bottom of Block
+     * On mobile — Toolbar at the bottom of Block
+     * On Desktop — Toolbar should be moved to the first line of block text
+     *              To do that, we compute the block offset and the padding-top of the plugin content
      */
-    if (!isMobile) {
-      const contentOffset = Math.floor(blockHeight / 2);
-
-      this.nodes.plusButton.style.transform = `translate3d(0, calc(${contentOffset}px - 50%), 0)`;
-      this.Editor.Toolbox.nodes.toolbox.style.transform = `translate3d(0, calc(${contentOffset}px - 50%), 0)`;
+    if (isMobile) {
+      toolbarY = targetBlockHolder.offsetTop + blockHeight;
     } else {
-      toolbarY += blockHeight;
+      toolbarY = targetBlockHolder.offsetTop + blockRenderedElementPaddingTop;
     }
 
     /**
      * Move Toolbar to the Top coordinate of Block
      */
     this.nodes.wrapper.style.transform = `translate3D(0, ${Math.floor(toolbarY)}px, 0)`;
+
+    /**
+     * Plus Button should be shown only for __empty__ __default__ block
+     */
+    if (block.tool.isDefault && block.isEmpty) {
+      this.plusButton.show();
+    } else {
+      this.plusButton.hide();
```

Comment:
Might be a good idea to move plus button into toolbox as well

Reply:
I don't think that it would be better. Toolbox is the list of buttons. It is about to become vertical. And it can be appeared in several ways, not only by Plus Button. And the Plus Button is just a part of Toolbar UI.